File: selfstudy/golang/players_counthigh.go
/*
task 267:
Players - CountHigh
You need to count how many values are greater than 50 in player heights in cm
*/

package main

import (
	"fmt"
	"reflect"
)

func Count_above_fiftyPlayers(heights []int) int {
	count := 0
	for _, v := range heights {
		if v > 50 { count++ }
	}
	return count
}

func main() {
	tests := []struct {
		input []int
		expected int
	}{
		{[]int{10, 20, 30}, Count_above_fiftyPlayers([]int{10, 20, 30})},
		{[]int{50, 60, 70}, Count_above_fiftyPlayers([]int{50, 60, 70})},
		{[]int{1, 2, 3}, Count_above_fiftyPlayers([]int{1, 2, 3})},
		{[]int{100, 200}, Count_above_fiftyPlayers([]int{100, 200})},
		{[]int{}, Count_above_fiftyPlayers([]int{})},
	}
	
	for idx, t := range tests {
		res := Count_above_fiftyPlayers(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// finally works

File: selfstudy/js/players_counthigh.js
/*
task 67:
Players - CountHigh
You need to count how many values are greater than 50 in player heights in cm
*/

function count_above_fifty_players(heights) {
  return heights.filter(x => x > 50).length;
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: count_above_fifty_players([10, 20, 30]) },
  { input: [50, 60, 70], expected: count_above_fifty_players([50, 60, 70]) },
  { input: [1, 2, 3], expected: count_above_fifty_players([1, 2, 3]) },
  { input: [100, 200], expected: count_above_fifty_players([100, 200]) },
  { input: [], expected: count_above_fifty_players([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('count_above_fifty_players', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(count_above_fifty_players(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, count_above_fifty_players(t.input) === t.expected || JSON.stringify(count_above_fifty_players(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// works i think
